File: csharp/Game/CharacterFactory.cs
namespace Game
{
    public static class CharacterFactory
    {
        // In C#, the factory method uses a switch expression, which is more concise than the Java switch-case syntax.
        // Additionally, C# uses 'throw new ArgumentException' for error handling, while Java uses 'throw new IllegalArgumentException'.
        public static Character Create(string type, string name, int health) =>
            type.ToLower() switch
            {
                "player" => new Player(name, health),
                "enemy" => new Enemy(name, health),
                _ => throw new ArgumentException($"Unknown type: {type}")
            };
    }
}

File: java/game/src/main/java/example/CharacterFactory.java
package example;

// This class showcases the Factory design pattern in Java.
// It demonstrates the use of a static method to create objects based on a type parameter.
public class CharacterFactory {

    private CharacterFactory() {
    }
    public static Character create(String type, String name, int health) {
        switch (type.toLowerCase()) {
            case "player" -> {
                return new Player(name, health);
            }
            case "enemy" -> {
                return new Enemy(name, health);
            }
            default -> throw new IllegalArgumentException("Unknown type: " + type);
        }
    }
}

